plugins/inputs/mesos/mesos.go

```diff
@@ -593,11 +672,119 @@ func (m *Mesos) gatherMainMetrics(u *url.URL, role Role, acc telegraf.Accumulato
 		}
 	}
 
+	var includeTaggedFields bool
+	for _, col := range m.MasterCols {
+		if col == "framework_offers" || col == "allocator" {
+			includeTaggedFields = true
+			break
+		}
+	}
+
+	taggedFields := map[string][]TaggedField{}
+	extraTags := map[string]map[string]string{}
+
+	if includeTaggedFields {
+		for metricName, val := range jf.Fields {
+			if !strings.HasPrefix(metricName, "master/frameworks/") && !strings.HasPrefix(metricName, "allocator/") {
+				continue
+			}
+
+			parts := strings.Split(metricName, "/")
+			if (parts[0] == "master" && len(parts) < 5) || (parts[0] == "allocator" && len(parts) <= 5) {
```

Comment:
Why we have `< 5` and `<= 5` later code expect `len(parts) > 5`. I prefer to make it more explicit with a comment why we need 5 parts.

Reply:
"framework offers" metrics all have at least 5 parts, and "allocator" metrics with 5 or less parts can be sent through without pulling any extra tags. I've added that in a comment to clarify.

You can peruse the [allocator metrics in the docs here](http://mesos.apache.org/documentation/latest/monitoring/#allocator) to see the different metric names, where some don't require any parts pulled out into tags e.g. `allocator/mesos/allocation_run_ms/count` vs `allocator/mesos/roles/<role>/shares/dominant` where `<role>` is pulled out into a `roles: <role>` tag and then omitted from resulting metric name `allocator/mesos/roles/shares/dominant`